packages/snap/src/index.ts

```diff
@@ -1,33 +1,92 @@
-import { OnRpcRequestHandler } from '@metamask/snaps-types';
-import { panel, text } from '@metamask/snaps-ui';
+import { providerErrors, rpcErrors } from '@metamask/rpc-errors';
+import { DialogType, OnRpcRequestHandler } from '@metamask/snaps-types';
+import { copyable, heading, panel, text } from '@metamask/snaps-ui';
+import { SLIP10Node } from '@metamask/key-tree';
+import {
+  add0x,
+  assert,
+  bytesToHex,
+  remove0x,
+  valueToBytes,
+} from '@metamask/utils';
+import { sign as signEd25519 } from '@noble/ed25519';
+import { sign as signSecp256k1 } from '@noble/secp256k1';
+
+import type { GetBip32PublicKeyParams, SignMessageParams } from './types';
 
 /**
  * Handle incoming JSON-RPC requests, sent through `wallet_invokeSnap`.
  *
  * @param args - The request handler args as object.
- * @param args.origin - The origin of the request, e.g., the website that
- * invoked the snap.
  * @param args.request - A validated JSON-RPC request object.
- * @returns The result of `snap_dialog`.
+ * @returns The result of the method.
  * @throws If the request method is not valid for this snap.
  */
-export const onRpcRequest: OnRpcRequestHandler = ({ origin, request }) => {
+export const onRpcRequest: OnRpcRequestHandler = async ({ request }) => {
   switch (request.method) {
-    case 'hello':
-      return snap.request({
+    case 'getPublicKey':
+      return await snap.request({
+        method: 'snap_getBip32PublicKey',
+        params: request.params as unknown as GetBip32PublicKeyParams,
+      });
+
+    case 'signMessage': {
+      const { message, curve, ...params } = request.params as SignMessageParams;
```

Comment:
This destructuring doesn't look like it's behaving as expected. The message isn't being displayed correctly in the snap, and signing seems to be failing
<img width="1015" alt="Screenshot 2023-10-04 at 7 44 33 PM" src="https://github.com/Sovereign-Labs/sov-snap/assets/32944016/34c24a44-2660-4912-9e3e-8ea30053af33">

Reply:
The domain of frontend development hehehe. I developed this under Firefox Beta, and it worked. I'll check for a more compatible way, as this is part of the snap

Reply:
I tested the last commit with chromium and it worked. Did you use a `Metamask Flask` for the local test site? Vanilla Metamask doesn't have the functionality required to start the site - however, it works fine with the Snap

Reply:
Yep, this was with `Metamask Flask` on Chrome